### PR TITLE
API doc cleanup for decorators

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
Makes descriptions parallel, cleans up styles and formats to be consistent.
Mostly for decorators, but some other API issues that were reported.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/di/interface/provider.ts [hunk 1]
```diff
@@ -123,37 +117,35 @@ export interface StaticClassProvider extends StaticClassSansProvider {
  */
 export interface ConstructorSansProvider {
   /**
-   * A list of `token`s which need to be resolved by the injector. The list of values is then
-   * used as arguments to the `useClass` constructor.
+   * A list of `token`s  to be resolved by the injector.
```
Comment (IgorMinar) [inline packages/core/src/di/interface/provider.ts]:
```suggestion
   * A list of `token`s to be resolved by the injector.
```

File: packages/core/src/di/interface/provider.ts [hunk 2]
```diff
@@ -162,62 +154,56 @@ export interface ConstructorProvider extends ConstructorSansProvider {
 /**
  * Configures the `Injector` to return a value of another `useExisting` token.
  *
- * For more details, see the ["Dependency Injection Guide"](guide/dependency-injection).
+ * @see ["Dependency Injection Guide"](guide/dependency-injection).
```
Comment (IgorMinar) [inline packages/core/src/di/interface/provider.ts]:
```suggestion
 * @see `ExistingProvider`
 * @see ["Dependency Injection Guide"](guide/dependency-injection).
```

File: packages/core/src/di/interface/provider.ts [hunk 3]
```diff
@@ -162,62 +154,56 @@ export interface ConstructorProvider extends ConstructorSansProvider {
...
  *
  * @usageNotes
  *
- * ### Example
- *
- * {@example core/di/ts/provider_spec.ts region='ExistingSansProvider'}
+ * {@example core/di/ts/provider_spec.ts region='ExistingSansProvider' linenums="false"}
```
Comment (IgorMinar) [inline packages/core/src/di/interface/provider.ts]: does this region actually exist? I don't see it defined in the provider_spec.ts file.

I just noticed that `ConstructorProvider` is not exported from @angular/core at all (this makes it inaccessible to developers for direct interaction but allows type inference and typechecking to work just fine - meaning that developers would usually not complain about the lack of exporting but the docs for this symbol are missing on angular.io)

can you please fix this my exporting the symbol from packages/core/src/di/index.t and update tools/public_api_guard/core/core.d.ts to reflect the adition?

File: packages/core/src/linker/component_factory_resolver.ts [hunk 4]
```diff
@@ -34,10 +34,19 @@ class _NullComponentFactoryResolver implements ComponentFactoryResolver {
 }
 
 /**
+ * Stores factory classes that have been defined to create components of specific types.
```
Comment (IgorMinar) [inline packages/core/src/linker/component_factory_resolver.ts]:
```suggestion
 * A simple registry that maps `Components` to generated `ComponentFactory` classes that can be used to create instances of components.
```

**Review by IgorMinar — APPROVED** [review]
great improvements! thank you

I left a few comments / suggestions.

File: packages/core/src/metadata/di.ts [hunk 5]
```diff
@@ -188,33 +189,34 @@ export const ContentChildren: ContentChildrenDecorator = makePropDecorator(
...
-   * * **static** - whether or not to resolve query results before change detection runs (i.e.
-   * return static results only). If this option is not provided, the compiler will fall back
-   * to its default behavior, which is to use query results to determine the timing of query
-   * resolution. If any query results are inside a nested view (e.g. *ngIf), the query will be
-   * resolved after change detection runs. Otherwise, it will be resolved before change detection
+   * * **selector** - The directive type or the name used for querying.
+   * * **read** - True to read a different token from the queried element.
+   * * **static** - True to resolve query results before change detection runs.
+   *
+   * When `static` is false or not provided, uses query results to determine the timing of query
```
Comment (devversion) [inline packages/core/src/metadata/di.ts]: This seems to be wrong. If the `static` flag is set to `false`, the query timing is not determined automatically.

Comment (devversion) [inline packages/core/src/metadata/di.ts]: This seems wrong to me. If the `static` flag is set to `false`, the query timing is not resolved automatically. The results will never be available before change detection ran (at least based on https://github.com/angular/angular/pull/28810)

File: packages/core/src/di/interface/provider.ts (hunk 3 above)
Comment (jbogarthyde) [reply packages/core/src/di/interface/provider.ts]: I can update the index, but I don't have access to the golden file.  I've asked George to do it.

Comment (gkalpak) [reply packages/core/src/di/interface/provider.ts]: I foudn out there were a few more missing, so added them all: https://github.com/angular/angular/pull/31377/commits/6b5bf085d6b9b00a266bb1a9982df302e3f49867

**Comment by jbogarthyde** [comment]
Please fix the CLA and re-run codefresh.

**Pull request closed without merging**